src/components/ui/Arc.tsx

```diff
@@ -0,0 +1,181 @@
+/**
+
+1. **コンポーネントの概要**
+- `RotatingArcs`という3Dシーン用のUIコンポーネントを実装しています
+- サイバーパンク風のターゲティング/スキャンエフェクトを表示する機能を持っています
+
+2. **主な機能**
+- 以下の3つの主要なアニメーション状態を管理します：
+  - スタンバイ状態（通常状態）
+  - ターゲティング状態（移動アニメーション）
+  - スキャン状態（ロック完了状態）
+
+3. **アニメーションの詳細**
+- ターゲティングアニメーション：
+  - 3つのキーポイントを経由する複雑な移動パスを実装
+  - 左上→右上→左下→中心という順序で動作
+  - イージング関数を使用して滑らかな動きを実現
+
+4. **視覚的要素**
+- 複数の回転するアーク（弧）
+- ターゲティングレティクル（照準）
+- スキャンエフェクト
+- ステータス表示（"STANDBY"、"TARGETING..."、"TARGET LOCKED"）
+
+5. **タイミング制御**
+- 5秒ごとにアニメーションサイクルを実行
+- ターゲティング（2秒）→スキャン（3秒）の順で動作
+
+ */
+
+'use client';
+
+import { useEffect, useState, useRef, useCallback } from 'react';
+import { Html } from '@react-three/drei';
+import styles from '@/components/ui/arc.module.css';
+
+export default function RotatingArcs({
+  position = [0, 0, 0] as [number, number, number],
+  rotation = [0, 0, 0] as [number, number, number],
+}) {
+  const [scanning, setScanning] = useState(false);
+  const [targeting, setTargeting] = useState(false);
+  const [offset, setOffset] = useState({ x: 0, y: 0 });
+  const animationRef = useRef<number | null>(null);
+  const startTimeRef = useRef(Date.now());
+
+  // 特定のパスに沿ったアニメーション
+  const animatePathMovement = useCallback(() => {
+    if (!targeting) return;
+
+    const elapsedTime = Date.now() - startTimeRef.current;
+    const totalDuration = 2000;
+    const progress = Math.min(elapsedTime / totalDuration, 1);
+
+    // 3つのキーポイントを定義
+    const keyPoints = [
+      { x: -80, y: -60 },
+      { x: 100, y: -80 },
+      { x: -80, y: 60 },
+      { x: 0, y: 0 },
+    ];
```

Comment:
[imo]
この keyPoints の上にこんな感じ(↓)の定数を定義して、`if (progress < 1 / 3)`の部分を`if (progress < stage.FIRST)`とか`if (progress < stage.SECOND)`みたいな感じにすると、とりあえず1個目の位置にいるかどうかがみたいのかな、2個目の位置にいるかをみたいのかなって推察しやすくなるかもです！

```suggestion
    // アニメーションの各段階を定数として定義
    const stage = {
      FIRST: 1 / 3, // 左上 → 右上
      SECOND: 2 / 3, // 右上 → 左下
      THIRD: 1, // 左下 → 中心
    };

    // 3つのキーポイントを定義
    const keyPoints = [
      { x: -80, y: -60 }, // 左上
      { x: 100, y: -80 }, // 右上
      { x: -80, y: 60 }, // 左下
      { x: 0, y: 0 }, // 中心
    ];
```